video/out/vo_gpu_next.c

```diff
@@ -956,10 +958,11 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
             .interpolation_threshold = opts->interpolation_threshold,
         );
 
-        // mpv likes to generate sporadically jumping PTS shortly after
-        // initialization, but pl_queue does not like these. Hard-clamp as
-        // a simple work-around.
-        qparams.pts = p->last_pts = MPMAX(qparams.pts, p->last_pts);
+        // When using interpolation, the given pts may possibly be less than
+        // the lowest value in pl_queue which it does not. Hard-clamp as a
+        // simple work-around.
+        if (can_interpolate)
+            qparams.pts = p->last_pts = MPMAX(qparams.pts, p->last_pts);
```

Comment:
```c
struct pl_source_frame first;
if (pl_queue_peek(p->queue, 0, &first)) {
    qparams.pts = MPMAX(qparams.pts, first.pts);
}
```

Reply:
Nice, that seems to work perfectly.

Reply:
Pulled this bit out into a separate commit.